### PR TITLE
Document requirements on kubernetes clone for cherry-picking.

### LINKED ISSUE #7106: hack/cherry-pick.sh fails with "9da202e can not be resolved as commit".
Following https://github.com/openshift/origin/blob/master/HACKING.md#cherry-picking-an-upstream-commit-into-origin, this does not work:

```sh
$ hack/cherry-pick.sh 19600

++ pr/19600 appears to have merged at fc2929ed05a8500b7ea5b990b843fa1543b7efb3
++ Generating patch for 5914deeac82e5ad5d288517aa0aaaee5aa5b381e..e90de3f9855b68ffe4dbbbbd98484c2a3ec0e113 onto 9da202e ...
fatal: Cannot update paths and switch to branch 'last_upstream_branch' at the same time.
Did you intend to checkout '9da202e' which can not be resolved as commit?
```

`9da202e` comes from

```
go run /home/jsafrane/project/go/src/github.com/openshift/origin/tools/godepversion/godepversion.go /home/jsafrane/project/go/src/github.com/openshift/origin/Godeps/Godeps.json k8s.io/kubernetes/pkg/api
```

There is no commit `9da202e` in Kubernetes upstream.

### PR DESCRIPTION
**Pull request by jsafrane**
Fixes #7106.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -202,8 +202,15 @@ If you are not updating packages you should not need godep installed.
 You can use `hack/cherry-pick.sh` to generate patches for Origin from upstream commits. To use
 this command, be sure to setup remote branches like https://gist.github.com/piscisaureus/3342247
 so that `git show origin/pr/<number>` displays information about your branch after a `git fetch`.
-You must also have the Kubernetes repository checked out in your GOPATH (visible as `../../../k8s.io/kubernetes`)
-and have no modified or uncommitted files in either repository.
+You must also have the Kubernetes repository checked out in your GOPATH (visible as `../../../k8s.io/kubernetes`),
+with openshift/kubernetes as a remote and fetched:
+
+    $ pushd ../../../k8s.io/kubernetes
```
Comment (liggitt) [inline HACKING.md]: `$GOPATH/src/k8s.io/kubernetes`

**Comment by liggitt** [comment]
nit on path, LGTM otherwise

**Comment by liggitt** [comment]
alternately, hack/cherry-pick.sh could validate the k8s repository remotes... that would probably be better

File: HACKING.md (hunk 1 above)
Comment (jsafrane) [inline HACKING.md]: fixed

**Comment by liggitt** [comment]
Lgtm, [merge]

**Pull request merged**